tools/analyzer_plugin/lib/src/diagnostic/exhaustive_deps.dart

```diff
@@ -926,7 +926,7 @@ class ExhaustiveDeps extends DiagnosticContributor {
         }
 
         Expression? maybeID = declaredDependencyNode;
-        while (maybeID is PropertyAccess || maybeID is PrefixedIdentifier) {
+        while (maybeID != null && (maybeID is PropertyAccess || maybeID is PrefixedIdentifier)) {
```

Comment:
This `!= null` is redundant, but without it `maybeID` isn't promoted to be non-nullable